### ISSUE TITLE
Add semantic support for constrained parameters

### ISSUE THREAD

**Issue by brightly-salty** [brightly-salty/mlatu#10]
There is currently neither syntactic nor semantic support for constrained parameters.

An example:
```
define print (for t. -> +io):
  show Mlatu.print
```

In using `print`, there is an implicit requirement that `t` has an instance of `show`. Currently, it is a runtime error for `print` to be used on a `t` that does not have an instance of `show`.

The proposed syntax is something like the following.
```
define print (for t. t show => t -> +io:
  show Mlatu.print
```

*Edited for correctness and updating of syntax*

**Comment by brightly-salty** [comment]
Waiting on #21

**Comment by brightly-salty** [comment]
Closing in favor of GH Project-based workflow
